src/App/Pages/Accounts/LoginPageViewModel.cs

```diff
@@ -187,15 +188,40 @@ public async Task LogInAsync(bool showLoading = true)
                     LogInSuccessAction?.Invoke();
                 }
             }
+            // Cozy customization, intercept SanitizeUrl exceptions
+            //*
+            catch (CozyException e)
+            {
+                await _deviceActionService.HideLoadingAsync();
+                var translatedErrorMessage = AppResources.ResourceManager.GetString(e.GetType().Name, AppResources.Culture);
+                await _platformUtilsService.ShowDialogAsync(translatedErrorMessage, AppResources.AnErrorHasOccurred, AppResources.Ok);
+            }
+            //*/
             catch (ApiException e)
             {
                 _captchaToken = null;
                 MasterPassword = string.Empty;
                 await _deviceActionService.HideLoadingAsync();
                 if (e?.Error != null)
                 {
+                    // Cozy customization, set custom message for 401 response
+                    // As the stack does not translate error messages and 401 is the most common error
+                    // then we intercept this specific error to translate it on client side
+                    /*
                     await _platformUtilsService.ShowDialogAsync(e.Error.GetSingleMessage(),
                         AppResources.AnErrorHasOccurred, AppResources.Ok);
+                    /*/
+                    if (e.Error.StatusCode == HttpStatusCode.Unauthorized)
```

Comment:
info: Soon 401 can be also bad bearer token https://github.com/cozy/cozy-stack/pull/3173 aka expired token or else.

Reply:
Thanks for the info.

As we are calling login route (`/connect/token`) I suppose that receiving an expired token error will not be possible? Or is this a bad assumption?

Reply:
I think you're right but let's put @nono in ;)

Reply:
I confirm